tests/src/Kernel/CollaboraMediaAccessTest.php

```diff
@@ -0,0 +1,440 @@
+                'edit any document in collabora',
+            ]),
+            'Kelly (edit own)' => $this->createUser([
+                'edit own document in collabora',
+            ]),
+            'Media admin' => $this->createUser([
+                'administer media',
+            ]),
+        ];
+
+        /** @var \Drupal\media\MediaInterface[] $media_entities */
+        $media_entities = [
+            "Sean's published document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Sean (preview own)']->id(),
+            ]),
+            "Sean's unpublished document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Sean (preview own)']->id(),
+                'status' => 0,
+            ]),
+            "Kelly's published document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Kelly (edit own)']->id(),
+            ]),
+            "Kelly's unpublished document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Kelly (edit own)']->id(),
+                'status' => 0,
+            ]),
+        ];
+
+        $this->assertEntityAccess(
+            [
+                'anonymous' => [],
+                'authenticated' => [],
+                'media permissions only' => [],
+                'Bookworm' => [],
+                'Previewer' => [
+                    "Sean's published document" => ['preview in collabora'],
+                    "Kelly's published document" => ['preview in collabora'],
+                ],
+                'Sean (preview own)' => [
+                    "Sean's unpublished document" => ['preview in collabora'],
+                ],
+                'Editor' => [
+                    "Sean's published document" => ['edit in collabora'],
+                    "Sean's unpublished document" => ['edit in collabora'],
+                    "Kelly's published document" => ['edit in collabora'],
+                    "Kelly's unpublished document" => ['edit in collabora'],
+                ],
+                'Kelly (edit own)' => [
+                    "Kelly's published document" => ['edit in collabora'],
+                    "Kelly's unpublished document" => ['edit in collabora'],
+                ],
+                'Media admin' => [
+                    "Sean's published document" => ['preview in collabora', 'edit in collabora'],
+                    "Sean's unpublished document" => ['preview in collabora', 'edit in collabora'],
+                    "Kelly's published document" => ['preview in collabora', 'edit in collabora'],
+                    "Kelly's unpublished document" => ['preview in collabora', 'edit in collabora'],
+                ],
+            ],
+            $accounts,
+            $media_entities,
+            [
+                'preview in collabora',
+                'edit in collabora',
+            ],
+        );
+
+        $this->assertEntityPathsAccess(
```

Comment:
Any thoughts on which permission combinations we should check with paths, specifically?
If we want sufficient coverage but no duplication.

Some options I can think of:
- Check a few permissions with paths, but not the full range.
- Check the same permissions with paths as with entity access calls, but hide/automate the duplication, e.g. in an assert method that checks paths and entity access using the same data.
- Implement a custom entity access hook in a test method, connected to a service that is remote-controlled from the test.
- Mock the access control handler for media.

Reply:
Usually is a dedicated test only for that, like this: https://git.drupalcode.org/project/file_defaults_by_type/-/blob/1.x/modules/file_defaults_by_type_ui/tests/src/Kernel/RouteAccessTest.php

That would mean duplicate code indeed, since you want to cover all the cases. Or you create some code to provide data or you do the second point.

Second option seems the more sensible for me since the code is there and ready.